src/main/java/com/google/devtools/build/lib/rules/cpp/CppHelper.java

```diff
@@ -949,7 +949,8 @@ public static String getDLLHashSuffix(
             ruleContext.getConfiguration().getOptions().get(CppOptions.class));
     if (cppOptions.renameDLL
         && cppOptions.dynamicMode != DynamicMode.OFF
-        && featureConfiguration.isEnabled(CppRuleClasses.TARGETS_WINDOWS)) {
+        && featureConfiguration.isEnabled(CppRuleClasses.TARGETS_WINDOWS)
+        && ruleContext.isAttrDefined("win_def_file", LABEL)) {
```

Comment:
I don't think that's enough, you have to check
`&& (!ruleContext.isAttrDefined("win_def_file", LABEL) || ruleContext.getPrerequisiteArtifact("win_def_file") == null)`

Please also put a comment here to explain why we need it.